sdks/java/core/src/main/java/org/apache/beam/sdk/io/Source.java

```diff
@@ -64,7 +64,7 @@
   /** @deprecated Override {@link #getOutputCoder()} instead. */
   @Deprecated
   public Coder<T> getDefaultOutputCoder() {
-    throw new UnsupportedOperationException("Source needs to override getOutputCoder()");
+    return getOutputCoder();
```

Comment:
Unfortunately I don't see how to avoid this in a backward-compatible way.

* People need to be able to override either of these methods (old implementations override old one, new implementations override new one), so neither of them can be abstract.
* People also need to be able to call either of these methods, so neither of them can have a default body of "throw new UnsupportedOperationException", because then someone who calls that version might call someone else who overrides the other version.